src/templates/home.js

```diff
@@ -0,0 +1,115 @@
+import React, { useState, useMemo, useCallback } from 'react'
+import { useColorMode, Flex, Stack, IconButton, Button } from '@chakra-ui/core'
+
+import { STATUS, useLoadMore } from '../hooks/useLoader'
+import { useSearch } from '../hooks/useSearch'
+
+import Layout from '../components/layout'
+import SEO from '../components/seo'
+import { List } from '../components/list'
+import { Card } from '../components/card'
+import { Search } from '../components/search'
+import { Banner } from '../components/banner'
+import { Row } from '../components/row'
+import { Title } from '../components/title'
+
+function sortByDate(order, list) {
+  if (order === 'asc') {
+    return [...list].reverse()
+  }
+
+  return list
+}
+
+const IndexPage = ({ pageContext }) => {
+  const { data, pageInfo, status, loadNextPage } = useLoadMore({
+    initialData: pageContext.nodes,
+    initialPageInfo: pageContext.pageInfo
+  })
+  const { colorMode, toggleColorMode } = useColorMode()
+  const { onSearch, results, term } = useSearch({ data })
+  const [sort, setSort] = useState(() => 'desc')
+
+  const handleChangeSort = useCallback(
+    ({ target }) => setSort(() => target.value),
+    []
+  )
+
+  const persons = useMemo(
+    () => ((term.length > 2) ? results : data),
+    [results, data, term]
+  )
+
+  const isLoading = status === STATUS.loading
+  const shouldRenderLoadMoreButton = pageInfo.hasNextPage && term.length < 3
+
+  const backgroundButtonColor = { dark: 'blue.100', light: 'blue.200' }
+  const colorButton = { dark: 'black', light: 'white' }
+  const stateButton = { dark: 'blue.200', light: 'blue.100' }
+
```

Comment:
É uma boa prática manter os valores dentro de `useMemo`'s. Caso o componente seja re-renderizado, o valor também não será redeclarado sem necessidade. Por exemplo os valores `colorButton` e `stateButton` se encaixam neste exemplo.
Agora, o `isLoading` deve ser redeclarado sempre que o `status` for atualizado. Isso mudará apenas o valor de `isLoading`, sem o `useMemo` o componente seria inteiramente re-renderizado

https://reactjs.org/docs/hooks-reference.html#usememo
```suggestion

  const isLoading = useMemo(() => status === STATUS.loading, [status])
  const shouldRenderLoadMoreButton = useMemo(() => pageInfo.hasNextPage && term.length < 3, [pageInfo, term])

  const backgroundButtonColor = useMemo(() => ({ dark: 'blue.100', light: 'blue.200' }), [])
  const colorButton = useMemo(() => ({ dark: 'black', light: 'white' }), [])
  const stateButton = useMemo(() => ({ dark: 'blue.200', light: 'blue.100' }), [])

```